notes/stv/20220509-user-impersonate-sh-01.txt

```diff
@@ -0,0 +1,232 @@
+bash: /home/fedora/zeppelin-0.10.0-bin-all/conf/zeppelin-env.sh: Permission denied
+Error: Could not find or load main class org.apache.zeppelin.interpreter.remote.RemoteInterpreterServer
+
+	at org.apache.zeppelin.interpreter.remote.ExecRemoteInterpreterProcess.start(ExecRemoteInterpreterProcess.java:97)
+	at org.apache.zeppelin.interpreter.ManagedInterpreterGroup.getOrCreateInterpreterProcess(ManagedInterpreterGroup.java:68)
+	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.getOrCreateInterpreterProcess(RemoteInterpreter.java:104)
+	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.internal_create(RemoteInterpreter.java:154)
+	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.open(RemoteInterpreter.java:126)
+
+	... 13 more
+
+
+
+# Create zeppelinusers group, and add fedora and new user to it
+
+  groupadd zeppelinusers
+  usermod -a -G zeppelinusers gaiadmp
+  usermod -a -G zeppelinusers fedora
+
+
+# Change permissions of fedora/zeppelin folders
+
+  sudo chmod 750 fedora/
+  sudo chmod 750 -R fedora/zeppelin-0.10.0-bin-all/
+  sudo chown -R fedora:zeppelinusers fedora/
+ 
+
+# Set home directory to each user's
+
+# Open sh interpreter in Zeppelin UI
+
+Set:
+ shell.working.directory.user.home	true
+
+
+	
+# Restart Zeppelin
+
+/home/fedora/zeppelin-0.10.0-bin-all/bin/zeppelin-daemon.sh restart
+
+
+
+# New sh notebook..
+
+%sh
+whoami
+
+> gaiadmp
+
+
+%sh
+whoami
+
+> /home/gaiadmp
+
+
+%sh
+ls /home/fedora/zeppelin-0.10.0-bin-all
+
+>
+
+bin
+conf
+interpreter
+k8s
+lib
+LICENSE
+licenses
+local-repo
+logs
+notebook
+notebook2
+NOTICE
+plugins
+README.md
+run
+scripts
+spark-warehouse
+webapps
+zeppelin-web-0.10.0.war
+zeppelin-web-angular-0.10.0.war
+
+
+%sh
+cat /home/fedora/zeppelin-0.10.0-bin-all/conf/shiro.ini
+
+# Shows full shiro.ini file
+# Not good, so any user would have access to zeppelin configuration files and thus credentials to our local SQL database
+
+# So summary so far, user impersonate works, but only if the user has read/execute access to the zeppelin configuration & parent directory data (i.e. /fedora/..)
+
+
+# Let's try restricting the permission of sensitive files
+
+
+%sh
+cat /home/fedora/zeppelin-0.10.0-bin-all/conf/shiro.ini
+
+> .. Permission Denied
+
+
+
+
+
+# After some experimentation, it seems like the following is the minimum amount of permissions we need to give "zeppelin users"
+
+sudo chown fedora:zeppelinusers /home/fedora/
+sudo chown fedora:zeppelinusers /home/fedora/zeppelin-0.10.0-bin-all
+sudo chown -R fedora:zeppelinusers /home/fedora/zeppelin-0.10.0-bin-all/interpreter
+sudo chown fedora:zeppelinusers /home/fedora/zeppelin-0.10.0-bin-all/conf
+sudo chown fedora:zeppelinusers /home/fedora/zeppelin-0.10.0-bin-all/conf/zeppelin-env.sh
```

Comment:
This is really important to know. I've created a new issue to find a way of keeping track of things like this.